core/ast.h

```diff
@@ -432,21 +432,21 @@ struct Function : public AST {
     }
 };
 
-struct LiteralString;
-
 /** Represents import "file". */
 struct Import : public AST {
-    LiteralString *file;
-    Import(const LocationRange &lr, const Fodder &open_fodder, LiteralString *file)
+    // Must be a LiteralString.
```

Comment:
well, yes, but you can handle that while keeping it a *LiteralString, instead of:
```
desugar(ast->file, obj_level);
```
you could:
```
AST *file = ast->file;
desugar(file, obj_level);
ast->file = dynamic_cast<LiteralString *>(ast_);
```
I think it's better, because:
a) It's more explicit - it clearly states that it needs to be LiteralString, but in this case desugarer could potentially put something else there.
b) In general a little bit nicer data structures and a little less nice code within some functions is better than the other way around, imo.